packages/flutter/lib/src/material/elevated_button.dart

```diff
@@ -182,9 +191,11 @@ class ElevatedButton extends ButtonStyleButton {
     final MaterialStateProperty<Color?>? foregroundColorProp = (foreground == null && disabledForeground == null)
       ? null
       : _ElevatedButtonDefaultColor(foreground, disabledForeground);
-    final MaterialStateProperty<Color?>? overlayColor = (foreground == null)
+    final MaterialStateProperty<Color?>? overlayColorProp = (foreground == null && overlayColor == null)
       ? null
-      : _ElevatedButtonDefaultOverlay(foreground);
+      : overlayColor != null && overlayColor.value == 0
+        ? const MaterialStatePropertyAll<Color?>(Colors.transparent)
+        : _ElevatedButtonDefaultOverlay((overlayColor ?? foreground)!);
```

Comment:
I found this double `?:` a little hard to follow here. I wonder if a switch expression could make this more readable?

```dart
final MaterialStateProperty<Color?>? overlayColorProp = switch((foreground, overlayColor)) {
  (null, null) => null,
  (_, Color overlayColor) when overlayColor.value == 0 => const MaterialStatePropertyAll<Color?>(Colors.transparent),
  (_, _) => _ElevatedButtonDefaultOverlay((overlayColor ?? foreground)!),
};
```

(same elsewhere in this PR where the same pattern is used)

Reply:
That's MUCH better. I really disliked the complicated `?:` but didn't understand how to apply the new pattern matching switch statement to them. Thanks for the pointer!